File: ios/VIRGINSApp/Sources/Models/Match.swift
import Foundation

struct Match: Codable, Identifiable {
    let id: String
    let user: MatchedUser
    let matchedAt: Date
    let conversationId: String?

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case user, matchedAt, conversationId
    }
}

struct MatchedUser: Codable, Identifiable {
    let id: String
    var displayName: String
    var profileImages: [String]
    var trustLevel: Int
    var isOnline: Bool
    var location: String?

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case displayName, profileImages, trustLevel, isOnline, location
    }

    var profileImage: String { profileImages.first ?? "" }
}

struct DiscoverProfile: Codable, Identifiable {
    let id: String
    let user: User
    let score: Double
    let reasons: [String]
    let breakdown: ScoreBreakdown

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case user, score, reasons, breakdown
    }
}

struct ScoreBreakdown: Codable {
    let faithScore: Double
    let valuesScore: Double
    let intentionScore: Double
    let lifestyleScore: Double
    let locationBonus: Double
    let trustBonus: Double
}

File: android/VIRGINSApp/app/src/main/java/app/virgins/data/model/Match.kt
package app.virgins.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Match(
    @Json(name = "_id") val id: String,
    val userId1: String,
    val userId2: String,
    val matchedAt: String,
    val status: String,
    val partner: MatchedUser? = null
)

@JsonClass(generateAdapter = true)
data class MatchedUser(
    @Json(name = "_id") val id: String,
    val displayName: String,
    val profileImages: List<String> = emptyList(),
    val trustLevel: Int = 1,
    val isOnline: Boolean = false,
    val denomination: String? = null,
    val age: Int? = null
)

@JsonClass(generateAdapter = true)
data class DiscoverProfile(
    val user: User,
    val score: Int,
    val reasons: List<String> = emptyList(),
    val breakdown: ScoreBreakdown? = null
)

@JsonClass(generateAdapter = true)
data class ScoreBreakdown(
    val faith: Int,
    val values: Int,
    val intention: Int,
    val lifestyle: Int,
    val location: Int,
    val trustBonus: Double
)

@JsonClass(generateAdapter = true)
data class LikeResponse(
    val matched: Boolean,
    val conversationId: String? = null
)
